tgui/packages/tgui/interfaces/GeneTek.js

```diff
@@ -0,0 +1,1316 @@
+    Emitter: true,
+    Reclaimer: true,
+  };
+  for (const { label, cooldown } of equipmentCooldown) {
+    haveDevice[label] = true;
+    onCooldown[label] = cooldown > 0;
+  }
+
+  const dnaGood = dna.every(pair => !pair[2]);
+  const dnaGoodExceptLocks = dna.every(pair =>
+    !pair[2] || pair[3] === "locked");
+
+  return (
+    <Section
+      title={name}
+      buttons={
+        <GeneIcon
+          name={icon}
+          size={1.5} />
+      }>
+      {!!booth && booth.ref === ref && (
+        <Modal>
+          <Section
+            width={35}
+            title={name}
+            style={{
+              "margin": "-10px",
+              "margin-right": "2px",
+            }}
+            buttons={(
+              <GeneIcon
+                name={icon}
+                size={4}
+                style={{
+                  "margin-top": "-2px",
+                  "margin-right": "-4px",
+                }} />
+            )}>
+            <LabeledList>
+              <LabeledList.Item label="Price">
+                <NumberInput
+                  minValue={0}
+                  maxValue={999999}
+                  width={5}
+                  value={booth.price}
+                  onChange={(_, price) => setBooth({
+                    ref: booth.ref,
+                    price: price,
+                    desc: booth.desc,
+                  })} />
+              </LabeledList.Item>
+              <LabeledList.Item label="Description">
+                <Input
+                  width={25}
+                  value={booth.desc}
+                  onChange={(_, desc) => setBooth({
+                    ref: booth.ref,
+                    price: booth.price,
+                    desc: desc,
+                  })} />
+              </LabeledList.Item>
+            </LabeledList>
+            <Box
+              inline
+              width="50%"
+              textAlign="center"
+              mt={2}>
+              <Button
+                icon="person-booth"
+                color="good"
+                disabled={boothCost > materialCur}
+                onClick={() => act("booth", booth)}>
+                Send to Booth
+              </Button>
+            </Box>
+            <Box
+              inline
+              width="50%"
+              textAlign="center">
+              <Button
+                icon="times"
+                color="bad"
+                onClick={() => setBooth(null)}>
+                Cancel
+              </Button>
+            </Box>
+          </Section>
+        </Modal>
+      )}
+      <UnlockModal />
+      <Box textAlign="right">
+        <Box
+          mr={1}
+          style={{ "float": "left" }}>
+          <Icon
+            color={research >= 3 ? "good" : research >= 2 ? "teal" : research >= 1 ? "average" : "bad"}
+            name={research >= 2 ? "flask" : research >= 1 ? "hourglass" : "times"} />
+          {research >= 2 ? " Researched" : research >= 1 ? " In Progress" : " Not Researched"}
```

Comment:
Prefer a mapping of number -> object with relevant properties for this (like the occupant status in the Sleeper).